packages/features/schedules/lib/use-schedule/useSchedule.ts

```diff
@@ -29,24 +29,31 @@ export const useSchedule = ({
   return trpc.viewer.public.slots.getSchedule.useQuery(
     {
       usernameList: getUsernameList(username ?? ""),
-      eventTypeSlug: eventSlug!,
+      // Prioritize slug over id, since slug is the first value we get available.
+      // If we have a slug, we don't need to fetch the id.
+      // TODO: are queries using eventTypeId faster? Even tho we lost time fetching the id with the slug.
+      ...(eventSlug ? { eventTypeSlug: eventSlug } : { eventTypeId: eventId ?? 0 }),
       // @TODO: Old code fetched 2 days ago if we were fetching the current month.
       // Do we want / need to keep that behavior?
       startTime: monthDayjs.startOf("month").toISOString(),
       // if `prefetchNextMonth` is true, two months are fetched at once.
       endTime: (prefetchNextMonth ? nextMonthDayjs : monthDayjs).endOf("month").toISOString(),
       timeZone: timezone!,
-      eventTypeId: eventId!,
       duration: duration ? `${duration}` : undefined,
     },
     {
+      trpc: {
+        context: {
+          skipBatch: true,
+        },
+      },
       refetchOnWindowFocus: false,
       enabled:
         Boolean(username) &&
-        Boolean(eventSlug) &&
-        (Boolean(eventId) || eventId === 0) &&
         Boolean(month) &&
-        Boolean(timezone),
+        Boolean(timezone) &&
+        // Should only wait for one or the other, not both.
```

Comment:
<img width="1019" alt="Screenshot 2023-08-07 at 10 01 14 AM" src="https://github.com/calcom/cal.com/assets/1780212/67358e0d-6211-45e8-a1ae-d3ae26563b30">
It might be breaking dynamic booking.

Reply:
Shouldn't a failed test for dynamic pop up here? I think we have that.

Reply:
Dynamic is working fine because it needs `eventSlug` and due to whatever reasons `eventSlug` is there always, but regular bookings availability are broken.
See https://github.com/calcom/cal.com/pull/10607/files#r1286568174

Reply:
Regular event types working fine for me. Also dynamics.